src/v/cluster/tests/ephemeral_credential_test.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "cluster/ephemeral_credential_frontend.h"
+#include "cluster/feature_manager.h"
+#include "cluster/tests/cluster_test_fixture.h"
+#include "cluster/types.h"
+#include "features/feature_table.h"
+#include "model/metadata.h"
+#include "security/acl.h"
+#include "test_utils/async.h"
+#include "test_utils/fixture.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/sleep.hh>
+#include <seastar/util/defer.hh>
+#include <seastar/util/log.hh>
+
+#include <algorithm>
+#include <iterator>
+#include <system_error>
+#include <vector>
+
+using namespace std::chrono_literals;
+using namespace cluster;
+
+FIXTURE_TEST(test_ephemeral_credential_frontend, cluster_test_fixture) {
+    std::vector<application*> apps{
+      create_node_application(model::node_id{0}),
+      create_node_application(model::node_id{1}),
+      create_node_application(model::node_id{2})};
+
+    wait_for_all_members(3s).get();
+
+    tests::cooperative_spin_wait_with_timeout(10s, [&apps] {
+        return apps.front()->controller->get_feature_table().local().is_active(
+          features::feature::ephemeral_secrets);
+    }).get();
+
+    ephemeral_credential_frontend& fe_0
+      = apps[0]->controller->get_ephemeral_credential_frontend().local();
+    ephemeral_credential_frontend& fe_1
+      = apps[1]->controller->get_ephemeral_credential_frontend().local();
+
+    security::acl_principal principal{
+      security::principal_type::ephemeral_user, "principal"};
+
+    // Put credentials on broker 0
+    auto cred_0 = cluster::impl::make_credentials(principal);
+    fe_0.put(cred_0).get();
+
+    // Test get credentials on broker 0
+    auto get_res = fe_0.get(principal).get();
+    BOOST_REQUIRE_EQUAL(errc::success, get_res.err);
+    BOOST_REQUIRE_EQUAL(cred_0, get_res.credential);
+
+    // Get credentials from broker 1, expect different to broker 0
+    auto get_res_1 = fe_1.get(principal).get();
+    BOOST_REQUIRE_EQUAL(errc::success, get_res_1.err);
+    BOOST_REQUIRE_NE(cred_0, get_res_1.credential);
+
+    // Check inform works
+    {
+        auto res = fe_0.inform(apps[1]->controller->self(), principal).get();
```

Comment:
The CI failure here is due to `rpc - connection_cache.h:90 - No client available`, is there something else I should wait on?

Reply:
`_self` was not initialised. Fixed.